cluster/shard_mapper.go

```diff
@@ -146,8 +148,8 @@ func (r *RemoteMapper) Open() (err error) {
 	return nil
 }
 
-func (r *RemoteMapper) SetRemote(m tsdb.Mapper) error {
-	return fmt.Errorf("cannot set remote mapper on a remote mapper")
+func (r *RemoteMapper) SetRemote(m tsdb.Mapper) {
```

Comment:
minor: I know it's outside the scope of this refactor but the fact that this func exists really bothers me.  I'm assuming it's only here to satisfy an interface, which again, doesn't feel right.